hashicorp/dev-portal
# style: reorg tutorial metadata

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/consul/tutorials/vault-secure/vault-pki-consul-secure-tls) 🔎
- [Asana task](https://app.asana.com/0/home/1199634971443477/1204604036286170) 🎟️
- [Figma file 🎨 ](https://www.figma.com/file/NAFjIPkbwjyY8jPl6T9nlP/Tutorial-Variants?type=design&node-id=297-36193&t=PlF0QTBCVhYIMs2w-0)

## 🗒️ What

This PR restyles the tutorial metadata area in preparation for the addition of the variant picker UI, which will render below the metadata area. The layout, spacing, and color of the entire tutorial metadata area has been reworked to match[ this new spec](https://www.figma.com/file/NAFjIPkbwjyY8jPl6T9nlP/Tutorial-Variants?type=design&node-id=297-36193&t=PlF0QTBCVhYIMs2w-0). These changes are independent of variants and can be released on their own, as there will still be many tutorials without variants, so the metadata changes are global despite variant state. 

<img width="767" alt="Screenshot 2023-05-15 at 5 36 28 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/f1fee097-e5b8-4281-a4db-81bfbbae533e">

## 📸 Design Screenshots
<img width="696" alt="Screenshot 2023-05-16 at 10 28 42 AM" src="https://github.com/hashicorp/dev-portal/assets/36613477/11e64de2-81b2-4652-acc4-fa68c6eacb11">
<img width="346" alt="Screenshot 2023-05-16 at 10 29 26 AM" src="https://github.com/hashicorp/dev-portal/assets/36613477/52a8dd3b-c214-4205-be0a-537706eec8e8">

## 🧪 Testing

- [Visit a tutorial ](https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/consul/tutorials/vault-secure/vault-pki-consul-connect-ca)with lots of metadata, interactive lab, various badge types etc 
- Ensure that it works in dark and light theme.
- Ensure it scales between mobile / desktop
- Test on all browsers
- Log in and test the authenticated state
- Bookmark and un-bookmark a tutorial to validate that the icons update as expected.
- More examples: 
- https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/nomad/tutorials/integrate-vault/vault-postgres
- https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/nomad/tutorials/autoscaler/dynamic-application-sizing
- https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/consul/tutorials/resiliency/introduction-chaos-engineering
- [Visit the library](https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/tutorials/library) and choose different combinations of metadata to validate that all variations work. 

## 💭 Anything else?
